tldw_Server_API/app/api/v1/endpoints/auth.py

```
@@ -244,6 +309,272 @@ def _normalize_magic_email(email: str) -> str:
+        ttl_seconds=int(auth_request["ttl_seconds"]),
+    )
+    return RedirectResponse(url=str(auth_request["auth_url"]), status_code=status.HTTP_307_TEMPORARY_REDIRECT)
+
+
+@router.get("/federation/{provider_slug}/callback", name="federation_callback", response_model=TokenResponse)
+async def federation_callback(
+    provider_slug: str,
+    code: str,
+    state: str,
+    request: Request,
+    db=Depends(get_db_transaction),
```

Ruff (0.15.6)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)